README.md

```diff
@@ -1,2 +1,841 @@
+
+**Note:** Format section ensures the ITN DID Format complies with [W3C DID Method Syntax](https://w3c.github.io/did-core/#method-syntax)
+
+### Operations
+
+Identity Protocol provides methods to manage DIDs.
+
+#### Usage of Identity Protocols
+
+```typescript
+import { IdentityProtocol, Agent } from "@itn-trust/agent"
+
+const identityProtocol = new IdentityProtocol({
+  "create-did": async (message: CreateDIDMessage, next) => {
+    // handle CreateDIDMessage
+  },
+  "create-did-response": async (message: CreateDIDResponseMessage, next) => {
+    // handle CreateDIDResponseMessage
+  },
+  "update-did": async (message: UpdateDIDMessage, next) => {
+    // handle UpdateDIDMessage
+  },
+  "update-did-response": async (message: UpdateDIDResponseMessage, next) => {
+    // handle UpdateDIDResponseMessage
+  },
+  "revoke-did": async (message: RevokeDIDMessage, next) => {
+    // handle RevokeDIDMessage
+  },
+  "revoke-did-response": async (message: RevokeDIDResponseMessage, next) => {
+    // handle RevokeDIDResponseMessage
+  },
+  "recover-did": async (message: RecoverDIDMessage, next) => {
+    // handle RecoverDIDMessage
+  },
+  "recover-did-response": async (message: RecoverDIDResponseMessage, next) => {
+    // handle RecoverDIDResponseMessage
+  },
+})
+
+const agent = new Agent({
+  // ...
+  protocols: [identityProtocol],
+})
+```
+
+**Refer** [ITN Identity Protocol](https://github.com/itn-trust/itn/tree/master/packages/sdk/agent/src/protocol/itn.mobi/identity/1.0) for technical details. `DELETE LATER BEFORE CREATING PR`
+
+---
+
+#### Create
+
+Description: Based on the DID data schema context file and DID method an ITN process anchors a DID and its DID document and delivers it to the requestor.
+Constraints: Compliance with the DID data schema and DID method requirements. Can only be invoked by the DID controller(s).
+
+
+##### `create()` API
+
+```ts
+create({
+  creator: DIDString | DIDDocument,
+  newDID?: {
+    didDoc: DIDDocument,
+    recoveryKey: Ed25519VerificationKey2020,
+  }
+}): Promise<DIDDocument>
+```
+* `creator` - creator of the new DID, can be a DID string or DID document.
+* `newDID.didDoc` - (optional) a pre-created DID Document.
+* `newDID.recoveryKey` - (optional) a pre-created DID Document's recovery key.
+
+Sends `CreateDIDMessage` to `creator` and receives `CreateDIDResponseMessage` message in `create-did-response` handler.
+
+**example:**
+
+```typescript
+await agent.protocol(IdentityProtocol).create({ creator: "did:itn:TW7PbLSe2Ws8FCx675oFUU" })
+```
+
+---
+
+#### Resolve
+
+Description: Based on the DID data schema context file and DID method an ITN process delivers a DID document in accordance with the W3C DID standard.
+Constraints: Compliance with the DID data schema context file and DID method requirements. Any entity should be able to invoke the process.
+
+##### `getDIDDoc()` API
```

Comment:
`getDIDDoc()` is not a DID Resolution. It simply loads DID Document from the Wallet. To resolve DID, the [agent.resolveDIDDoc()](https://github.com/itn-trust/itn/tree/master/packages/sdk/agent#resolvediddoc) function must be used.

Reply:
@bparth24 same here: this is just for explanation purposes, it doesn't mean we should include such implementation details.

Reply:
Noted and updated.